Document agentic usage attribution in Audit Trail (#37955)
DataDog/documentation: content/en/account_management/audit_trail/guides/track_agentic_usage_in_your_organization.md

---
title: Track Agentic Usage in Your Organization
description: Use Audit Trail to see which actions Datadog AI agents take in your organization, distinguish agent activity from human activity, and trace agent actions back to the user who authorized them.
disable_toc: false
further_reading:
- link: "account_management/audit_trail/"
  tag: "Documentation"
  text: "Set up Audit Trail"
- link: "account_management/audit_trail/events/"
  tag: "Documentation"
  text: "Learn about Audit Trail events"
---

## Overview

Datadog AI agents, including the Bits fleet (Bits SRE, Bits Dev, Bits Assistant, Bits Security Analyst) and Agent Builder, can take actions inside your organization on their own or on behalf of a user. Audit Trail attributes these actions to the agent that performed them and, when applicable, to the user who authorized them. This guide walks you through finding agent-driven events, filtering them by agent type, and tracing an agent's actions back to the user who delegated them.

## View all Datadog AI agent activity

1. Navigate to [Audit Trail][1].
2. In the search bar, enter the query: `@evt.actor.mode:*`.

>    This query returns every event where a Datadog AI agent is the actor, whether the agent acted `interactive` (a user was present and approved the action) or `autonomous` (the agent acted on a system trigger, with no user present).

Reviewer: ```suggestion
   This query returns every event where a Datadog AI agent is the actor, whether the agent acted `interactive` (a user was present and approved the action) or `autonomous` (the agent acted on a system trigger, with no user involved).
```

3. Click an event to open details and see the full actor breakdown: the delegator (the user, if any), the agent, and the mode.

**Note**: Rollout to all Bits agents is in progress, and coverage is not yet complete. Actions from agents that have not yet onboarded do not appear in `@evt.actor.mode:*` results.

## View actions taken through the Datadog MCP server

Actions taken through the Datadog MCP server, carry the `is_mcp_call` field.

1. Navigate to [Audit Trail][1].
2. In the search bar, enter the query: `is_mcp_call:true`.
3. Click an event to see a link back to the MCP server configuration and to other events from the same MCP session.

## View a user's full activity, including agent-delegated actions

> When a Datadog AI agent acts on a user's behalf, the agent becomes `@evt.actor` for that event, and the user moves to `@delegator.actor`. Searching only `@usr.name:<USER_EMAIL>` returns the actions the user performed directly, and does not include actions a Datadog AI agent performed on their behalf.

Reviewer: We might want to be a little more specific and mention that the agent now fills in the values of `@usr.*`, while the original user fills in `@delegator.*`? This should be easier to filter on than only the `@delegator.actor` field.

To retrieve a user's complete activity, including delegated agent actions:

1. Navigate to [Audit Trail][1].

> 2. In the search bar, enter the query: `(@usr.name:<USER_EMAIL> OR @delegator.name:<USER_EMAIL>)`. Replace `<USER_EMAIL>` with the user's email address.

Reviewer: `@delegator.name` is the actual full name. If users want to filter on the email address, I would advise `@delegator.id`.

## View all Datadog AI agent activity

1. Navigate to [Audit Trail][1].
2. In the search bar, enter the query: `@evt.actor.mode:*`.

   This query returns every event where a Datadog AI agent is the actor, whether the agent acted `interactive` (a user was present and approved the action) or `autonomous` (the agent acted on a system trigger, with no user involved).
3. Click an event to open details and see the full actor breakdown: the delegator (the user, if any), the agent, and the mode.

**Note**: Rollout to all Bits agents is in progress, and coverage is not yet complete. Actions from agents that have not yet onboarded do not appear in `@evt.actor.mode:*` results.

## View actions taken through the Datadog MCP server

> Actions taken through the Datadog MCP server, carry the `is_mcp_call` field.

Reviewer 2: ```suggestion
Actions taken through the Datadog MCP server carry the `is_mcp_call` field.
```

1. Navigate to [Audit Trail][1].
2. In the search bar, enter the query: `(@usr.id:<USER_ID> OR @delegator.id:<USER_ID>)`. Replace `<USER_ID>` with the user's ID.

   **Note**: `@delegator.name` and `@usr.name` hold the user's full name, not their email address. To filter by a specific user, use `@delegator.id` and `@usr.id` instead.

   | Description of audit event                                                                          | Query in audit explorer                                     |
   | ------------------------------------------------------------------------------------------------------| ------------------------------------------------------------ |
   | Actions a user performed directly                                                                     | `@usr.id:<USER_ID>`                                           |
   | All actions attributed to a user, including actions a Datadog AI agent performed on their behalf      | `(@usr.id:<USER_ID> OR @delegator.id:<USER_ID>)`               |
3. See [Create a dashboard or a graph][2] if you want to put this information into a dashboard or graph.

## Distinguish agent events from human events

> Agent-driven events display a visual badge in the event list, so you can distinguish them from human-driven events without opening each one.

Reviewer 2: What does the badge look like? A brief description here would help.

Author: It looks like this @brett0000FF: 
<img width="1071" height="385" alt="image" src="https://github.com/user-attachments/assets/d2442ff8-426f-4c0f-8264-28800bcf8b20" />

I can add this screenshot to the doc if it helps!

Reviewer 2: Thanks! I don't think a screenshot is necessary (would be a pain to keep up-to-date 😄). But we could be more descriptive in the text like:
```suggestion
Agent-driven events are marked with the agent's name and avatar (for example, Bits Assistant) paired with the delegating user, labeled Co-authored by user + agent. This lets you distinguish agent activity from human activity without opening each event.
```
